backend/src/main/kotlin/trie/TrieSearch.kt

```diff
@@ -0,0 +1,194 @@
+package trie
+
+import org.apache.commons.text.similarity.LevenshteinDistance
+
+class TrieSearch(private val trie: Trie, search: String, private val splitWords: Boolean = false) {
```

Comment:
In der Klasse verwendest relativ oft in `!!` operator - glaubst könn ma des irgendwie umschreiben dass ma den vermeiden/nur nehmen wos unbedingt notwendig is und sonst vorher validieren und glei drauf reagieren?
der is immer relativ gefählrich für NPEs -> internal server errors find i

Reply:
ja tu ich, aber es is immer entweder vorher gechecked obs eh drinnen is oder von der logik her zwingend, also i glaub NPEs können keine entstehn. aber des is halt des doofe mit der map api, a wenn i vorher gechecked hab mit am .containsKey muss i nachher a !! machen.

Reply:
ok :) dann würd i sagen lass ma des ah :D